dali/base/dafdesc.cpp

```diff
@@ -1340,6 +1352,13 @@ class CFileDescriptor:  public CFileDescriptorBase, implements ISuperFileDescrip
             if (_interleaved)
                 *_interleaved = interleaved;
         }
+        const char * buffer = mb.toByteArray()+mb.getPos();
+        const char * historyStr = "History";
+        bool hasEnoughData = (mb.length() - mb.getPos()) > strlen(historyStr);
```

Comment:
Also needs a comment here explaining this is a kludge to preserve backward compatilibty and explain why serialization version # can't be used.

It looks like the existing SERIALIZATION_VERSION checking is flawed... you should be able to up the version as, but the check is hardwired to match not a min version, but a fixed version(s).